tests/cluster/tests/22-replica-in-sync.tcl

```diff
@@ -109,10 +109,23 @@ test "Replica in loading state is hidden" {
 }
 
 test "Check disconnected replica not hidden from slots" {
+    # We want to disconnect the replica, but keep it alive so it can still gossip
+
+    # Make sure that the replica will not be able to re-connect to the master
+    R $master_id config set requirepass asdf
+
     # Disconnect replica from primary
     R $master_id client kill type replica
+
     # Check master to have no replicas
     assert {[s $master_id connected_slaves] == 0}
+
```

Comment:
changing password is a good idea, but here, i think we can call `after 1000` or other things to let the replica gossip its state

Reply:
Calling `after 1000` is a lazy way out, and it leaves the test exposed to race conditions. I.e. What if 1000 isn't enough, and the replica doesn't gossip before the test completes? it would miss its purpose. 

Generally any constant delay is always a bad idea (unless we wait for the wall-clock to pass and a key to expire).

I don't know cluster that well, and if there's some metric we can probe to see if it did any gossip. 
And I don't know if we have any specific concern here about the damage gossip can do, or we just wanna get some due to paranoia of the unknown implications.

Reply:
I added a component to verify that the primary has received and updated message from the replica. I'm not really sure it's needed for correctness, but I thought it was good to have.